File: config/configManager.py
import yaml
import os

class ConfigManager:
    _instance = None

    def __init__(self, config_name=None):
        if not hasattr(self, 'config'):
            self.config = None
        
        if config_name:
            config_path = os.path.join(os.path.dirname(__file__), config_name)
            self.load_config(config_path)

    @classmethod
    def get_instance(cls, config_name=None):
        if cls._instance is None:
            cls._instance = ConfigManager(config_name)
        return cls._instance

    def load_config(self, file_path):
        try:
            with open(file_path, 'r', encoding='utf-8') as f:
                self.config = yaml.safe_load(f)
            print('配置文件加载成功')
        except Exception as error:
            print('加载配置文件时出错:', error)
            self.config = None

    def get(self, key):
        if not self.config:
            print('配置未加载，请先调用 load_config 方法')
            return None

        keys = key.split('.')
        value = self.config

        for k in keys:
            if isinstance(value, dict) and k in value:
                value = value[k]
            else:
                return None

        return value

# 导出单例
config_manager_instance = ConfigManager.get_instance("config.yaml")

File: config/configManager.ts
const yaml = require('js-yaml');
const fs = require('fs');
const path = require('path');

class ConfigManager {
  private static instance: ConfigManager;
  private config: any;

  private constructor(config_name?: string) {
    if(config_name){
      const config_path = path.resolve(__dirname, config_name);
      this.loadConfig(config_path);
    }else{
      this.config = null;
    }
  }

  public static getInstance(config_name?: string): ConfigManager {
    if (!ConfigManager.instance) {
      ConfigManager.instance = new ConfigManager(config_name);
    }
    return ConfigManager.instance;
  }

  /**
   * 加载 YAML 配置文件
   * @param filePath - YAML 文件的路径
   */
  public loadConfig(filePath: string): void {
    try {
      const fileContents = fs.readFileSync(filePath, 'utf8');
      this.config = yaml.load(fileContents);
      console.log('配置文件加载成功');
    } catch (error) {
      console.error('加载配置文件时出错:', error);
      this.config = null;
    }
  }

  /**
   * 通过 key 获取配置值
   * @param key - 配置项的 key，支持点号分隔的嵌套 key
   * @returns 配置值，如果未找到则返回 undefined
   */
  public get<T = any>(key: string): T | undefined {
    if (!this.config) {
      console.error('配置未加载，请先调用 loadConfig 方法');
      return undefined;
    }

    const keys = key.split('.');
    let value: any = this.config;

    for (const k of keys) {
      if (value && typeof value === 'object' && k in value) {
        value = value[k];
      } else {
        return undefined;
      }
    }

    return value as T;
  }
}

// 导出单例
const configManagerInstance = ConfigManager.getInstance("config.yaml");

export default configManagerInstance

